packages/wonder-blocks-dropdown/components/single-select.js

```diff
@@ -230,23 +247,39 @@ export default class SingleSelect extends React.Component<Props, State> {
         // item in the menu, use the placeholder.
         const menuText = selectedItem ? selectedItem.props.label : placeholder;
 
-        const items = this.getMenuItems();
-
-        const opener = (
+        const dropdownOpener = opener ? (
+            <DropdownOpener
+                onClick={this.handleClick}
+                disabled={numItems === 0 || disabled}
+                anchorRef={this.onHandleOpenerRef}
+                testId={testId}
+            >
+                {opener}
+            </DropdownOpener>
+        ) : (
             <SelectOpener
                 {...sharedProps}
-                disabled={items.length === 0 || disabled}
+                disabled={numItems === 0 || disabled}
                 id={id}
                 isPlaceholder={!selectedItem}
                 light={light}
                 onOpenChanged={this.handleOpenChanged}
                 open={this.state.open}
-                ref={this.handleOpenerRef}
+                ref={this.onHandleOpenerRef}
```

Comment:
nit: usually `onFoo` is used for props and `handFoo` is used for instance methods.  Combining `on` and `handle` should probably be avoided.

Reply:
Nice catch! I'm refactoring this part to remove the need of the `anchorRef` prop inside `DropdownOpener`.